src/client/dfuse/dfuse.h

```diff
@@ -0,0 +1,746 @@
+/**
+ * (C) Copyright 2016-2019 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+
+#ifndef __DFUSE_H__
+#define __DFUSE_H__
+
+#include <fuse3/fuse.h>
+#include <fuse3/fuse_lowlevel.h>
+
+#include <gurt/list.h>
+#include <gurt/hash.h>
+
+#include "dfuse_cnss.h"
```

Comment:
True, this is historic as we previously had the cnss application, and IOF and CPPR were dynamically loaded plugins.  There are therefore calls backwards and forwards between what are now dfuse_main and dfuse_core.  I've unwound a lot of the callbacks and types used here but I don't think it's sensible to go much further without the ability to run/test this code.  Again, this would be something that would be done early on as part of MS3.

Reply:
I've combined the header files, so there's no dfuse_cnss.h any more.